parsers/MN.py

```diff
@@ -15,104 +15,102 @@
+            message=f"Data request did not succeed: {response.status_code}",
         )
+    data = response.json()
 
-    if None in web_json.values():
+    # validate that the expected and required keys are present
+    received_keys = set(data.keys())
+    expected_keys = set(JSON_API_MAPPING.keys())
+    if not set(expected_keys).issubset(received_keys):
+        missing = expected_keys - received_keys
         raise ParserException(
             parser="MN.py",
-            message=f"Fetched values contain null. Fetched data: {web_json}.",
+            message=f"Fetched JSON didn't include expected keys: {missing}",
         )
 
-    # Then we can safely parse them
-    query_data = {}
-    for query_key, src_key in JSON_QUERY_TO_SRC.items():
-        if "time" in query_key:
-            # convert to datetime
-            query_data[query_key] = datetime.fromisoformat(web_json[src_key]).replace(
-                tzinfo=TZ
-            )
-        else:
-            # or convert to float, might also be string
-            query_data[query_key] = float(web_json[src_key])
-
-    return query_data
-
-
-def query(session: Session, logger: Logger, zone_key: ZoneKey) -> dict[str, Any]:
-    """
-    Query the JSON endpoint and parse it.
-    """
+    # pick out relevant values into a new dictionary with keys with our names
+    result = {JSON_API_MAPPING[k]: data[k] for k in expected_keys}
 
-    target_response: Response = session.get(NDC_GENERATION)
-
-    if not target_response.ok:
+    # ensure that relevant values are set
+    if None in result.values():
         raise ParserException(
             parser="MN.py",
-            message=f"Data request did not succeed: {target_response.status_code}",
+            message=f"Fetched values contain null. Fetched data: {result}.",
         )
 
-    # Read as JSON
-    response_json = target_response.json()
-    query_result = parse_json(response_json, logger, zone_key)
-
-    return query_result
+    # parse values
+    for k, v in result.items():
+        if k == "datetime":
+            result[k] = datetime.fromisoformat(v).replace(tzinfo=TZ)
+        else:
+            result[k] = float(v)
+    return result
 
 
 def fetch_production(
-    zone_key: ZoneKey,
+    zone_key: ZoneKey = ZoneKey("MN"),
     session: Session = Session(),
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
 ):
     if target_datetime:
         raise NotImplementedError("This parser is not yet able to parse past dates.")
 
-    query_data = query(session, logger, zone_key)
-
-    # Calculated 'unknown' production from available data (consumption, import, solar, wind, tpp).
-    # 'unknown' consists of 92.8% coal, 5.8% oil and 1.4% hydro as per 2020; sources: IEA and IRENA statistics.
-    query_data["leftoverMW"] = round(
-        query_data["consumptionMW"]
-        - query_data["importMW"]
-        - query_data["solarMW"]
-        - query_data["windMW"],
-        13,
-    )
```

Comment:
This calculation maps exactly to the `dts` value, which is presented in the website https://ndc.energy.mn/ as:
![image](https://github.com/user-attachments/assets/3af76418-3e10-402e-875f-7ff56f1789bb)

Due to this, I figure it should be used directly. If it should be coal or unknown is not in any way obvious in my mind - whats clear is that Mongolia is mostly using coal though, and they didn't mind reporting this as some kind of fossil power plant with the icon.